lib/sweet_xml.ex

```diff
@@ -468,11 +477,15 @@ defmodule SweetXml do
       ...>    ], true)
       [message: 'Message', ul: %{a: 'Two'}]
   """
-  def xmap(parent, mapping), do: xmap(parent, mapping, false)
+  def xmap(parent, mapping), do: xmap(parent, mapping, %{is_keyword: false})
+
+  def xmap(nil, _, %{is_optional: true}), do: nil
+
+  def xmap(parent, [], atom) when is_atom(atom), do: xmap(parent, [], %{is_keyword: atom})
```

Comment:
Note that this clause here ensures that the changes to xmap are not breaking. All current uses will pass without trouble.